Server/controllers/leave.controller.js

```diff
@@ -1,18 +1,58 @@
 "use strict";
+const leavesDataLayer = require('../datalayer/leave.datalayer');
 /**
  * Class representing leaves controller
  *
  * @class leavesController
  */
 class leavesController {
 
-    static getLeavesByUserId (req, res) {}
+    static async getLeavesByUserId (req, res) {
+        const userId = req.query.user;
+        let leavesList;
+        try {
+            leavesList = await leavesDataLayer.getLeavesByUserId(userId);
+        } catch (err) {
+            console.log(err.message);
+            return res.status(400);
+        }
+        return res.status(200).json(leavesList);
+    }
 
-    static createLeave (req, res) {}
+    static async createLeave (req, res) {
+        let createdLeave;
+        try {
+            createdLeave = await leavesDataLayer.createLeave(req.body);
```

Comment:
We shouldn't pass req.body -- Instead extract properties and create a new object before passing. Here we are assuming `req.body` has all the properties, we should check if all required data is there before passing to datalayer

Reply:
Extracted req.body to a new object and checked those values for undefined.